dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EntryResource.java

```diff
@@ -438,79 +444,130 @@ public Entry setDiscourseTopic(@ApiParam(hidden = true) @Parameter(hidden = true
+        final ProcessorProgress progress = new ProcessorProgress();
         while (!progress.done) {
             final TransactionHelper transactionHelper = new TransactionHelper(sessionFactory);
             transactionHelper.transaction(() -> {
-                final List<Tool> tools =
-                    this.toolDAO.findAllTools(progress.offset, LANGUAGE_VERSION_PROCESSOR_PAGE_SIZE);
-                if (tools.size() < LANGUAGE_VERSION_PROCESSOR_PAGE_SIZE) {
+                final List<T> list = loader.apply(progress.offset);
+                if (list.size() < PROCESSOR_PAGE_SIZE) {
                     progress.done = true;
                 }
-                progress.processedEntries += tools.size();
-                LOG.info("Executing {} tool language version updates starting at offset {}",
-                    LANGUAGE_VERSION_PROCESSOR_PAGE_SIZE, progress.offset);
-                progress.offset += LANGUAGE_VERSION_PROCESSOR_PAGE_SIZE;
-                tools.stream()
-                    .filter(tool -> tool.getDescriptorType().size() == 1) // Only support tools with 1 language
-                    .forEach(tool -> {
-                        final String descriptorLanguageText = tool.getDescriptorType().get(0);
-                        LanguageHandlerInterface languageHandlerInterface = LanguageHandlerFactory.getInterface(
-                            DescriptorLanguage.convertShortStringToEnum(descriptorLanguageText));
-                        tool.getWorkflowVersions().stream()
-                            .filter(version -> allVersions || version.getVersionMetadata().getDescriptorTypeVersions().isEmpty())
-                            .forEach(version -> {
-                                final String primaryPath;
-                                if (DescriptorLanguage.convertShortStringToEnum(descriptorLanguageText) == DescriptorLanguage.WDL) {
-                                    primaryPath = version.getWdlPath();
-                                } else { // We will not add new languages to Tool (perhaps to AppTool), so this is safe
-                                    primaryPath = version.getCwlPath();
-                                }
-                                readSourceFilesAndUpdate(languageHandlerInterface, version, primaryPath);
-                            });
-                    });
+                progress.processedEntries += list.size();
+                LOG.info("Executing {} updates starting at offset {}", PROCESSOR_PAGE_SIZE, progress.offset);
+                progress.offset += PROCESSOR_PAGE_SIZE;
+                processor.apply(list, allVersions);
             });
         }
-        LOG.info("Completed processing {} legacy tools", progress.processedEntries);
         return progress.processedEntries;
     }
 
+    Function<Integer, List<Workflow>> getWorkflows = (offset) -> workflowDAO.findAllWorkflows(offset, PROCESSOR_PAGE_SIZE);
```

Comment:
Are any of these used outside of this class? 
e.g. seems like processWorkflowsForLanguageVersions could be private for example